synapse/federation/units.py

```diff
@@ -98,7 +98,7 @@ class Transaction(JsonEncodedObject):
         "pdus",
     ]
 
-    def __init__(self, transaction_id=None, pdus=[], **kwargs):
+    def __init__(self, transaction_id=None, pdus: list = None, **kwargs):
```

Comment:
In general when we have an option which does not need to be provided It should be set to a type of `Optional[foo]`. There's quite a few place that this should be updated in this PR.

Reply:
If you could point that out, that'd be helpful, because I think otherwise i'd break some expectations, and i dont have a full knowledge where i can and where i cant

Reply:
Everywhere it defaults to `None` should be `Optional` IMO.

Reply:
they defaulted to mutable constructors (`{}`, `[]`) before, so i assume they werent meant to be optional, but should i make it so?

Reply:
That pretty much meant you did not need to provide it, we're now making that equivalent to `None`, so they're optional, yes.